### DISCUSSION TITLE
Issue with no known solution and possible idea to fix: Outer Wall Print Order (Contour vs Hole)

### DISCUSSION THREAD

**Discussion by DarthLeahcim** [OrcaSlicer/OrcaSlicer#4886] [Ideas]
# Problem
I've got a print order issue I can't seem to solve. I'm printing 4 objects that have a hollow interior to make them stackable:
![image](https://github.com/SoftFever/OrcaSlicer/assets/2687067/dfab34c6-291e-45e7-803f-b6707312f41d)

They are designed in such a way to reduce filament usage and to match my 0.8 nozzle. Each object edge is exactly 2 walls thick, but both are considered Outer walls because of this:
![image](https://github.com/SoftFever/OrcaSlicer/assets/2687067/1e11f451-bbbd-4f87-ab41-93d7bd1bbbf3)

The issue is I can't find a way to select which "outer wall" gets printed first. Object 1 and 2 print the "inner" wall first:
![image](https://github.com/SoftFever/OrcaSlicer/assets/2687067/cec196cb-00bc-42d6-9acd-c1d86876b313)

Object 3 and 4 print the "outer" wall first: 
![image](https://github.com/SoftFever/OrcaSlicer/assets/2687067/7c53ad8f-7c13-4259-94d6-f635f8c022fe)

This occurs on every layer except the first layer, in which the "outer" layer is printed first. In the first few layers, like half of the inner hexagons of the larger object are printed before the outer wall for some reason: 
![image](https://github.com/SoftFever/OrcaSlicer/assets/2687067/a574a5c7-4a93-4746-9213-7db852948a24)

Changing the Intra-Layer order doesn't have any effect. Even if object 4 is printed first (original order is 2-1-4-3, new order is 4-2-1-3), objects 3 and 4 still print the "outer" wall first. I really like the smoothness the Scarf seam provides so I'm using it on this model. The root cause of this issue is the further the nozzle travels, the worse the starting wall is (normal nozzle ooze issue maybe?). I've got the retraction dialed in super low so it looks really good when printing one object, but not so much when it has to travel long distance to a different object. It's not blobbing on the print, but rather the wall is starting late. This can be seen in the second picture below. Adding extra length to the unretract can help in theory, but causes issues with shorter travel distances. 

Printing the "inner" wall first hides this issue on the "inner" wall and the Scarf seam gives it a really nice exterior, even with wall-start issues on the "inner" wall. I had a mostly successful print last night. Object 1 looked decent, but once the other objects were finished (upper third), it looked nearly perfect (looks better in person):
![IMG_20240405_095504441_AE](https://github.com/SoftFever/OrcaSlicer/assets/2687067/ac8713b0-69d2-455c-8cd1-2680fb814424)
![IMG_20240405_095621955_AE](https://github.com/SoftFever/OrcaSlicer/assets/2687067/13f02edc-7808-42e5-bf81-e573f22fed3b)

In contrast to Object 3, which the "outer" wall was printed first (apparently need to change Seam position from Aligned to Nearest as well): 
![IMG_20240405_095917324_AE](https://github.com/SoftFever/OrcaSlicer/assets/2687067/f1becd97-ff20-4528-ab49-9694e5bc1538)
![IMG_20240405_095944425](https://github.com/SoftFever/OrcaSlicer/assets/2687067/a1f61119-e2a0-429b-b8a3-564f93053431)

# Proposed Solution
Add an option under "Walls printing order" titled "Outer wall print order" with the options of "Auto", "Contours First", and "Holes First". 

The Scarf Seam settings already can differentiate between Contours ("outer" wall) and Holes ("inner" wall) very well as the Scarf seam is only being applied to the "outer" wall correctly with "Contour" being selected for the Scarf joint seam (beta) setting. With my limited programming experience, I'd imagine this would be the bulk of the work that's already completed. 

If anyone knows an already existing solution to either the order issue or the late starting wall issue, I'd love to hear it. 

Config, Project, and G-Code files:
[Dragon 0.8.zip](https://github.com/SoftFever/OrcaSlicer/files/14886832/Dragon.0.8.zip)
Orca Slicer v2.0.0 Official release
Klipper versions:
![image](https://github.com/SoftFever/OrcaSlicer/assets/2687067/856bb0af-f288-4c4b-a9d4-1de057a43001)

Edited for clarification

**Comment by DARKFiB3R** [comment]
Have you managed to find any other ways of improving this situation?

**Reply by DarthLeahcim** [reply]
Not directly, no. If I print one object at a time, it mostly mitigates the issue of travel distance causing a late wall start, but larger objects still print in strange orders. Oddly, if I print only single or only double hex towers (objects 1-3 above) they all print the same; hole then contour ("inner" then "outer"). However, if I try to print only triple hex towers (3 in a straight line), it will change which is printed first depending on the object. Each time I sliced a single hex tower type, I would hit "Fill Bed with Copies" so the objects are identical.
